src/pages/projectSettings/tabs/EventLogTab/EventLogTab.tsx

```diff
@@ -38,11 +47,121 @@ export const EventLogTab: React.FC = () => {
     },
   });
 
+  type logEntry = {
+    timestamp: Date;
+    user: string;
+    before?: RepoEventSettingsFragment | ProjectEventSettingsFragment;
+    after?: RepoEventSettingsFragment | ProjectEventSettingsFragment;
+  };
+
+  const eventData = isRepo
+    ? repoEventData?.repoEvents?.eventLogEntries
+    : projectEventData?.projectEvents?.eventLogEntries;
+
+  return (
+    <div data-cy="event-log">
+      {((eventData as logEntry[]) || []).map(
+        ({ user, timestamp, before, after }) => (
+          /* @ts-expect-error */
+          <EventLogCard key={`event_log_${timestamp}`}>
+            <EventLogHeader user={user} timestamp={timestamp} />
+            <Table
+              data={getEventDiffLines(before, after)}
+              columns={[
+                <TableHeader
+                  key="key"
+                  label={<span data-cy="key">Property</span>}
+                  sortBy={(datum: EventDiffLine) => datum.key}
+                />,
+                <TableHeader
+                  key="before"
+                  label={<span data-cy="before">Before</span>}
+                  sortBy={(datum: EventDiffLine) => datum.before}
+                />,
+                <TableHeader
+                  key="after"
+                  label={<span data-cy="after">After</span>}
+                  sortBy={(datum: EventDiffLine) => datum.after}
+                />,
+              ]}
+            >
+              {({ datum }) => (
+                <Row key={datum.key} data-cy="event-log-table-row">
+                  <Cell>
+                    <StyledCell>{datum.key ?? ""}</StyledCell>
+                  </Cell>
+                  <Cell>
+                    <StyledCell>{getEventValue(datum.before)}</StyledCell>
+                  </Cell>
+                  <Cell>
+                    <StyledCell>{getEventValue(datum.after)}</StyledCell>
+                  </Cell>
+                </Row>
+              )}
+            </Table>
+          </EventLogCard>
+        )
+      )}
+    </div>
+  );
+};
+
+interface Props {
+  timestamp: Date;
+  user: string;
+}
+
+const EventLogHeader: React.VFC<Props> = ({ user, timestamp }) => (
+  <StyledHeader>
+    <H3>{getDateCopy(timestamp)}</H3>
+
+    <div> {user} </div>
+  </StyledHeader>
+);
+
+/* @ts-expect-error */
+const EventLogCard = styled(Card)`
+  margin-bottom: 48px;
+  padding: ${size.m};
+  width: 150%;
+`;
+
+const StyledCell = styled("div")`
+  word-break: break-all;
+  font-size: 12px;
+  font-family: Menlo, Monaco, Consolas, "Courier New", monospace;
+`;
+
+const StyledHeader = styled("div")`
+  padding-bottom: ${size.l};
+  padding-left: ${size.xxs};
+`;
+
+const getEventValue = (value: any) => {
+  if (value === true) {
+    return "true";
+  }
+  if (value === false) {
+    return "false";
+  }
+  if (value === null || value === undefined) {
+    return "";
+  }
+
+  if (value === "") {
+    return '""';
+  }
+
+  const splitArray = JSON.stringify(value).split(",");
+
   return (
```

Comment:
We should be consistent about this functions return type. Returning either a string or jsx not both.

Reply:
I need both though, depending on what it is... Otherwise the array is hard to read.